pkg/webhook/pod/tikv_util.go

```diff
@@ -29,38 +29,52 @@ import (
 	"k8s.io/klog"
 )
 
+type controllerDesc struct {
+	name      string
+	namespace string
+	kind      string
+}
+
 // checkFormerTiKVPodStatus would check all the former tikv pods whether their store state were UP during Upgrading
 // check need both  check former pod is ready ,store up, and no evict leader
-func checkFormerTiKVPodStatus(kubeCli kubernetes.Interface, tc *v1alpha1.TidbCluster, ordinal int32, set *apps.StatefulSet, storesInfo *pdapi.StoresInfo) error {
-
-	tcName := tc.Name
-	namespace := tc.Namespace
+func checkFormerTiKVPodStatus(kubeCli kubernetes.Interface, controllerDesc controllerDesc, ordinal, specReplicas int32, set *apps.StatefulSet, storesInfo *pdapi.StoresInfo) error {
+	controllerKind := controllerDesc.kind
+	controllerNamespace := controllerDesc.namespace
+	controllerName := controllerDesc.name
+	namespace := controllerDesc.namespace
 
-	for i := range helper.GetPodOrdinals(tc.Spec.TiKV.Replicas, set) {
+	for i := range helper.GetPodOrdinals(specReplicas, set) {
 		if i <= ordinal {
 			continue
 		}
-		podName := memberUtil.TikvPodName(tcName, i)
+		var podName string
+		switch controllerKind {
+		case v1alpha1.TiDBClusterKind:
+			podName = memberUtil.TikvPodName(controllerName, i)
+		case v1alpha1.TiKVGroupKind:
+			podName = memberUtil.TiKVGroupPodName(controllerName, i)
+		default:
+			return fmt.Errorf("")
```

Comment:
`controllerDesc.kind` is set in the code and has only two possible values, so this is unreachable too. add an unreachable comment or use `panic("unreachable")`, then people don't need to consider this part of code as it's unreachable